apollo-federation/src/sources/connect/expand/tests/schemas/expand/keys.yaml

```diff
@@ -1,30 +1,37 @@
-# rover supergraph compose --config apollo-federation/src/sources/connect/expand/tests/schemas/expand/keys.yaml > apollo-federation/src/sources/connect/expand/tests/schemas/expand/keys.graphql
-federation_version: =2.10.0-alpha.3
 subgraphs:
   one:
     routing_url: none
     schema:
       sdl: |
         extend schema
-          @link(url: "https://specs.apollo.dev/federation/v2.8", import: ["@key"])
+          @link(url: "https://specs.apollo.dev/federation/v2.10", import: ["@key"])
           @link(url: "https://specs.apollo.dev/connect/v0.1", import: ["@connect"])
         type Query {
           t(id: ID!): T
             @connect(                                                                                                   # expect `key: "id"`
               http: { GET: "http://localhost/ts/{$$args.id}" }
-              selection: "id id2"
+              selection: "id id2 unselected"
               entity: true
             )
-          t2(id: ID! id2: ID!): T
+          t2(id: ID!, id2: ID!): T
             @connect(                                                                                                   # expect `key: "id id2"`
               http: { GET: "http://localhost/ts/{$$args.id}?id2={$$args.id2}" }
-              selection: "id id2"
+              selection: "id id2 unselected"
+              entity: true
+            )
+          """ Uses the `unselected` field as a key, but doesn't select it """
+          unselected(unselected: ID!): T
```

Comment:
this is interesting! it makes sense for simple entity resolvers, but i think we'll have to have different rules for batch entity resolvers because we'll need to match the keys in the response to the entity references

Reply:
There might also be some REST APIs where the order is guaranteed to be the same, so we'll probably want a mode for that as well

Reply:
totally. i don't think that's something we can default to (i've yet to hear about one), but maybe we allow opting into that. i've been thinking that the `batch` argument is an object to support multiple styles of configuration, so a `batch: { assumeOrdered: true }` or something makes sense.